### PR TITLE
Updates to gRPC generation

### PR DESCRIPTION
**Pull request by markmandel**
Some cleanup post the last series of gRPC updates and changes for future gRPC updates.

* Cleanup the instructions for better clarity and operational ordering.
* Fix for allocation gRPC generation to reuse the gRPC version include in the build container image rather than unnecessarily build its own.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

**Comment by markmandel** [comment]
@Kalaiselvi84 do want to check with you if any of the changes to the release process need improvement in any way - let me know!

**Comment by Kalaiselvi84** [comment]
> do want to check with you if any of the changes to the release process need improvement in any way - let me know!

Nope! We are good with our release process steps.👍🏻

File: .github/ISSUE_TEMPLATE/kubernetes_update.md [hunk 1]
```diff
@@ -38,10 +38,12 @@ List of items to do for upgrading to {version_1} {version_2} {version_3}
...
-    - [ ] Update the version number in C++ Cmake scripts [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/CMakeLists.txt#L100) and [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/cmake/prerequisites.cmake#L34)
+    - [ ] Update the `grpc_release_tag` in the SDK [base image grpc version](https://github.com/googleforgames/agones/blob/main/build/includes/sdk.mk).
+    - [ ] Update the gRPC version number in C++ CMake scripts, `AGONES_GRPC_VERSION` [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/CMakeLists.txt)
+          and `gRPC_GIT_TAG` [here](https://github.com/googleforgames/agones/blob/main/sdks/cpp/cmake/prerequisites.cmake)
+    - [ ] Regenerate all client sdks: [make gen-all-sdk-grpc](https://github.com/googleforgames/agones/blob/main/build/README.md#make-gen-all-sdk-grpc) 
+          This can take an hour or so, as the above changes force a rebuild. Plan your day accordingly 😃.
+    - [ ] Regenerate allocated API endpoints: [make gen-allocation-grpc](https://github.com/googleforgames/agones/blob/main/build/README.md#make-gen-allocation-grpc)
```
Comment (Kalaiselvi84) [inline .github/ISSUE_TEMPLATE/kubernetes_update.md:46]: First, we need to run `make gen-allocation-grpc`, followed by the make gen-all-sdk-grpc target🤔
Do we need also mention to run `make build` target followed by the sdk.mk file update and rebuilding all the sdk docker images if it is required?

Comment (markmandel) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md:46]: > First, we need to run `make gen-allocation-grpc`, followed by the make gen-all-sdk-grpc target🤔 

No I deliberately switched the order to make `make gen-all-sdk-grpc` run first. It's the more important step of the two.


> Do we need also mention to run `make build` target followed by the sdk.mk file update and rebuilding all the sdk docker images if it is required?

There's no need to run `make build`, everything that needs to get rebuilt gets rebuilt automatically (the sdk and allocation grpc build images are actually separate from the main Go/Agones build image).

Comment (Kalaiselvi84) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md:46]: I am super confused, but if these steps work for the Kubernetes upgrade, then it's fine.😅

**Pull request merged**